src/dataflow-types/src/types.rs

```diff
@@ -508,7 +508,65 @@ pub struct SourceDesc {
     /// to the output of the source.
     pub operators: Option<LinearOperator>,
     pub bare_desc: RelationDesc,
-    pub persisted_name: Option<String>,
+    pub persist_desc: Option<SourcePersistDesc>,
```

Comment:
let's make this look more like the SourceConnectorConnector we discussed offline

Reply:
:+1:

Do you have any suggestion that is not `SourceConnectorConnector`? I'm having nightmares from Enterprise Java ... `ConnectorBuilderFactoryFactory` :see_no_evil: